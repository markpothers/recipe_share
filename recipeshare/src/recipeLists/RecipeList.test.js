import 'react-native'
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

import RecipeList from './RecipesList.js'
import { listProps } from '../../__mocks__/recipeList/mockCardContents.js'

const mockStore = configureStore([])

describe('Recipe List', () => {

	let mountedRecipeList
	let mockCallBack
	let store
	let navigation

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(() => {
		// console.log('runs before every test')

		mockCallBack = jest.fn()
		store = mockStore({
			recipes: {
				all: [listProps[0]],
				// chef: [listProps[0]],
				// chef_feed: [listProps[0]],
				// chef_liked: [listProps[0]],
				// chef_made: [listProps[0]],
				// global_ranks: [listProps[0]],
				// most_liked: [listProps[0]],
				// most_made: [listProps[0]]
			},
			recipes_details: {},
			loggedInChef: {},
			global_ranking: '',
			filter_settings: {},
			cuisine: 'Any',
			serves: 'Any',
		})
		navigation = {
			addListener: jest.fn()
		}
	})

	afterEach(() => {
		// console.log('runs after each test')
		mountedRecipeList.unmount()
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	// it('agrees that 1 = 1', () => {
	// 	expect(1).toEqual(1)
	// })

	// it('asyncly works', done => {
	// 	setTimeout(done, 100)
	// })

	// it('asyncly works using done()', done => {
	// 	setTimeout(() => {
	// 		console.log('I ran within the set Timeout while waiting for done()')
	// 		done()
	// 	}, 3000)
	// })

	// function sleep(time) {
	// 	return new Promise((resolve) => setTimeout(() => {
	// 		console.log('waiting in the promise')
	// 		resolve(true)
	// 	}
	// 	), time);
	// }

	// it('asyncly works using await', async () => {
	// 	let bool = false
	// 	bool = await sleep(5000)
	// 	expect(bool).toEqual(true)
	// })

	it('can be rendered when all recipes are already in the store', async () => {
		mountedRecipeList = renderer.create(
			<Provider
				store={store}
			>
				<RecipeList
					listChoice={"all"}
					route={{ name: 'All Recipes' }}
					mockCallBack={mockCallBack}
					navigation={navigation}
				/>
			</Provider>
		)
		const image = mountedRecipeList.toJSON()
		expect(image).toMatchSnapshot()
	})

	it('has all the expected like Buttons', async () => {
		// console.warn(listProps[0])
		mountedRecipeList = await renderer.create(
			<Provider
				store={store}
			>
				<RecipeList
					listChoice={"all"}
					route={{ name: 'All Recipes' }}
					mockCallBack={mockCallBack}
					navigation={navigation}
				/>
			</Provider>
		)
		// console.warn(mountedRecipeList)
		// mountedRecipeList.forceUpdate()
		// let root = mountedRecipeList.root
		// const instance = mountedRecipeList.getInstance()
		// console.warn(root)
		// const buttons = root.findAllByProps({testID: "likeButton"})
		// expect(buttons.length).toEqual(1)
	})

	// it('can be rendered with chef recipes', () => {
	// 	mountedRecipeList = renderer.create(<Provider store={store}><RecipeList listChoice={"chef"} route={{name: 'My Recipes'}}  mockCallBack={mockCallBack} /></Provider>)
	// 	const image = mountedRecipeList.toJSON()
	// 	expect(image).toMatchSnapshot()
	// })

	// it('can be rendered with the chefs feed', () => {
	// 	mountedRecipeList = renderer.create(<Provider store={store}><RecipeList listChoice={"chef_feed"} route={{name: 'My Feed'}} mockCallBack={mockCallBack} /></Provider>)
	// 	const image = mountedRecipeList.toJSON()
	// 	expect(image).toMatchSnapshot()
	// })

	// it('can be rendered with the chef_liked feed', () => {
	// 	mountedRecipeList = renderer.create(<Provider store={store}><RecipeList listChoice={"chef_liked"} route={{name: 'Recipes I Like'}} mockCallBack={mockCallBack} /></Provider>)
	// 	const image = mountedRecipeList.toJSON()
	// 	expect(image).toMatchSnapshot()
	// })

	// it('can be rendered shallowly', () => {
	//     mountedRecipeList = shallow((<Provider store={store}><RecipeList mockCallBack={mockCallBack}/></Provider>))
	//     const image = mountedRecipeList.toJSON()
	//     expect(image).toMatchSnapshot()
	// })

	// it('can be rendered shallowly', () => {
	// 	// mountedRecipeList = mount((<Provider store={store}><RecipeList mockCallBack={mockCallBack}/></Provider>))
	// 	mountedRecipeList = renderer.create(<Provider store={store}><RecipeList mockCallBack={mockCallBack} /></Provider>)
	// 	// mountedRecipeList = shallow(<RecipeList mockCallBack={mockCallBack}/>, {
	// 	//     wrappingComponent: props => <Provider store={store}>{props.children}</Provider>
	// 	// })
	// 	// let testButton = mountedRecipeList.findWhere((node) => node.prop('testID') === 'filterButton' )

	// 	console.log(mountedRecipeList)
	// 	// expect(mountedRecipeList.contains(<RecipeList/>)).toEqual(true)
	// 	let list = mountedRecipeList.root.findAllByProps({ testID: 'filterButton' })
	// 	console.log(list.length)
	// 	// expect(mountedRecipeList.contains('<Connect(RecipesList) mockCallBack={[Function: mockConstructor]} />')).toEqual(true)
	// })

	// it('can be rendered shallowly', () => {
	//     mountedRecipeList = shallow(<RecipeList mockCallBack={mockCallBack}/>, {wrappingComponent: (<Provider store={store}></Provider>)})
	//     // mountedRecipeList = shallow((<Provider store={store}><RecipeList mockCallBack={mockCallBack}/></Provider>))
	//     let testButton = mountedRecipeList.findWhere((node) => node.prop('testID') === 'filterButton' )
	//     console.log(testButton.length)
	//     // expect(mountedRecipeList.contains(<RecipeList/>)).toEqual(true)
	// })

	// it('has a filter button', () => {
	//     mountedRecipeList = shallow(<Provider store={store}><RecipeList mockCallBack={mockCallBack}/></Provider>)
	//     const instance = mountedRecipeList.root
	//     const filterButton = instance.findByType("TouchableOpacity")
	//     console.log(filterButton)
	// })

	// it('can be deeply rendered', () => {
	//     mountedRecipeList = renderer.create(<Provider store={store}><RecipeList mockCallBack={mockCallBack}/></Provider>)
	//     console.log(mountedRecipeList)
	// })

	// it('looks the same as it did previously', () => {
	//     const tree = mountedRecipeList.toJSON()
	//     expect(tree).toMatchSnapshot()
	// })

	// it('has some state', () => {
	//     let mountedRecipeList = shallow(<Provider store={store}><RecipeList/></Provider>)
	//     let recipeList = mountedRecipeList.find('RecipeList')
	//     console.log(recipeList.first())
	//     // expect(mountedRecipeList.first().state()).toEqual({ count: 0})
	// })

	// it('has a button for displaying the filter menu', () => {
	//             let mountedRecipeList = render(<RecipeList/>)

	//     let recipeList = mountedRecipeList.find('RecipeList')
	//     console.log(recipeList.length)
	//     // let filterButton = mountedRecipeList.findWhere((node) => node.prop('testID') === 'filterButton')

	//     // console.log(filterButton.length)
	//     // expect(filterButton.length).toBe(1)
	// })

	// it('the filter button causes a setState call when pressed', () => {
	//     let filterButton = mountedRecipeList.findWhere((node) => node.prop('testID') === 'filterButton')
	//     filterButton.first().props().onPress()
	//     expect(mockCallBack.mock.calls.length).toEqual(1)
	// })




	// it('contains some touchables', () => {
	//     let mountedRecipeList = shallow(<Provider store={store}><RecipeList/></Provider>)
	//     let touchables = mountedRecipeList('TouchableOpacity')
	//     expect(touchables.length).toBe(4)
	// })

});
