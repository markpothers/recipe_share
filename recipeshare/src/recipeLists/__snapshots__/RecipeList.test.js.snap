// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe List can be rendered when all recipes are already in the store 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 0,
        "height": 1334,
        "position": "absolute",
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff59b",
        "borderColor": "#104e01",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 47,
        "justifyContent": "center",
        "left": "45%",
        "position": "absolute",
        "top": "10%",
        "width": 47,
        "zIndex": 1,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#104e01"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "left": 1.25,
          "top": 1.7,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "zIndex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 3.335,
            "marginLeft": 0,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "yellow",
              "borderWidth": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#104e01",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "overflow": "hidden",
                "width": 750,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="default"
              maxFontSizeMultiplier={2}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Search for Recipes"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "minHeight": 80.03999999999999,
                  "paddingLeft": 15,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                  "width": 675,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      ListHeaderComponent={[Function]}
      data={
        Array [
          Object {
            "acknowledgement": "Creamy mascarpone cheese and custard layered between espresso and rum soaked house-made ladyfingers, topped with Valrhona cocoa powder.",
            "bread": false,
            "breakfast": false,
            "chef_commented": 0,
            "chef_id": 12,
            "chef_liked": 0,
            "chef_made": 0,
            "chef_shared": 0,
            "chefimage_url": "https://storage.googleapis.com/download/storage/v1/b/chef-avatars-ac1cff8d11c908f3cf8613121b61683221152250/o/59b9a3ad9aaf1fe5f9fd5c92c62461eaa88fea89.jpg?generation=1601673323651524&alt=media",
            "chicken": true,
            "comments_count": 4,
            "created_at": "2020-10-02T19:09:33.552Z",
            "cuisine": "Turkish",
            "dairy_free": false,
            "description": "Fresh parsley, Italian sausage, shallots, garlic, sun-dried tomatoes and mozzarella cheese in an all-butter crust. With a side of mixed fruits.",
            "dessert": true,
            "difficulty": 6,
            "dinner": false,
            "freezer_meal": true,
            "gluten_free": false,
            "hidden": false,
            "id": 1001,
            "image_url": null,
            "keto": false,
            "likes_count": 3,
            "listChoice": "chef",
            "lunch": false,
            "makes_count": 1,
            "name": "Recipe with no cover images - TEST:NORMAL",
            "paleo": true,
            "red_meat": true,
            "renderOfflineMessage": Array [],
            "salad": true,
            "seafood": false,
            "serves": "3",
            "shares_count": 2,
            "side": true,
            "soup": false,
            "time": "06:00",
            "updated_at": "2020-10-02T22:07:27.075Z",
            "username": "pothers",
            "vegan": true,
            "vegetarian": true,
            "weekend": false,
            "weeknight": false,
            "white_meat": false,
            "whole_30": true,
          },
        ]
      }
      disableVirtualization={false}
      extraData={Object {}}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      listKey={
        Array [
          Object {
            "acknowledgement": "Creamy mascarpone cheese and custard layered between espresso and rum soaked house-made ladyfingers, topped with Valrhona cocoa powder.",
            "bread": false,
            "breakfast": false,
            "chef_commented": 0,
            "chef_id": 12,
            "chef_liked": 0,
            "chef_made": 0,
            "chef_shared": 0,
            "chefimage_url": "https://storage.googleapis.com/download/storage/v1/b/chef-avatars-ac1cff8d11c908f3cf8613121b61683221152250/o/59b9a3ad9aaf1fe5f9fd5c92c62461eaa88fea89.jpg?generation=1601673323651524&alt=media",
            "chicken": true,
            "comments_count": 4,
            "created_at": "2020-10-02T19:09:33.552Z",
            "cuisine": "Turkish",
            "dairy_free": false,
            "description": "Fresh parsley, Italian sausage, shallots, garlic, sun-dried tomatoes and mozzarella cheese in an all-butter crust. With a side of mixed fruits.",
            "dessert": true,
            "difficulty": 6,
            "dinner": false,
            "freezer_meal": true,
            "gluten_free": false,
            "hidden": false,
            "id": 1001,
            "image_url": null,
            "keto": false,
            "likes_count": 3,
            "listChoice": "chef",
            "lunch": false,
            "makes_count": 1,
            "name": "Recipe with no cover images - TEST:NORMAL",
            "paleo": true,
            "red_meat": true,
            "renderOfflineMessage": Array [],
            "salad": true,
            "seafood": false,
            "serves": "3",
            "shares_count": 2,
            "side": true,
            "soup": false,
            "time": "06:00",
            "updated_at": "2020-10-02T22:07:27.075Z",
            "username": "pothers",
            "vegan": true,
            "vegetarian": true,
            "weekend": false,
            "weeknight": false,
            "white_meat": false,
            "whole_30": true,
          },
        ]
      }
      maxToRenderPerBatch={10}
      nestedScrollEnabled={true}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={1}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={16}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 93.38000000000001,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#104e01",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "marginBottom": 2.668,
                "marginTop": 2.668,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": "1%",
                  "marginTop": "1%",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 8,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "marginLeft": "2%",
                      "marginRight": "2%",
                      "opacity": 1,
                      "width": "96%",
                    }
                  }
                  testID="recipeNameButton"
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "color": "#505050",
                        "fontSize": 32.12572956370018,
                        "fontWeight": "bold",
                        "textAlign": "left",
                      }
                    }
                  >
                    Recipe with no cover images - TEST:NORMAL
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "marginLeft": "2%",
                      "marginRight": "2%",
                      "width": "96%",
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "color": "#505050",
                        "fontSize": 29.06613627191921,
                        "fontStyle": "italic",
                      }
                    }
                  >
                    Created by: 
                  </Text>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="chefNameButton"
                  >
                    <Text
                      maxFontSizeMultiplier={2}
                      style={
                        Object {
                          "color": "#505050",
                          "fontSize": 29.06613627191921,
                          "fontStyle": "italic",
                        }
                      }
                    >
                      pothers
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginLeft": "2%",
                      "marginRight": "2%",
                      "width": "96%",
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "color": "#505050",
                        "flex": 1,
                        "fontSize": 29.06613627191921,
                      }
                    }
                  >
                    Prep time: 
                    06:00
                  </Text>
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "color": "#505050",
                        "flex": 1,
                        "fontSize": 29.06613627191921,
                      }
                    }
                  >
                    Difficulty: 
                    6
                    /10
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 2,
                    "justifyContent": "center",
                    "opacity": 1,
                    "overflow": "hidden",
                  }
                }
                testID="chefImageButton"
              >
                <Image
                  source={
                    Object {
                      "uri": "https://storage.googleapis.com/download/storage/v1/b/chef-avatars-ac1cff8d11c908f3cf8613121b61683221152250/o/59b9a3ad9aaf1fe5f9fd5c92c62461eaa88fea89.jpg?generation=1601673323651524&alt=media",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 5,
                      "height": 120.06,
                      "marginRight": "4%",
                      "width": 120.06,
                    }
                  }
                />
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 500.25,
                  "opacity": 1,
                  "width": 750,
                }
              }
              testID="recipeImageButton"
            >
              <Image
                source={1}
                style={
                  Object {
                    "borderRadius": 5,
                    "height": "100%",
                    "position": "absolute",
                    "width": 746,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": "1%",
                  "marginTop": "1%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID="reShareButton"
              >
                <Icon
                  name="share-outline"
                  size={46.690000000000005}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#505050",
                      "marginRight": "10%",
                    }
                  }
                />
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Object {
                      "color": "#505050",
                      "fontSize": 32.12572956370018,
                      "fontWeight": "bold",
                      "marginLeft": "5%",
                      "textAlign": "center",
                    }
                  }
                >
                  2
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID="likeButton"
              >
                <Icon
                  name="heart-outline"
                  size={46.690000000000005}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#505050",
                      "marginRight": "10%",
                    }
                  }
                />
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Object {
                      "color": "#505050",
                      "fontSize": 32.12572956370018,
                      "fontWeight": "bold",
                      "marginLeft": "5%",
                      "textAlign": "center",
                    }
                  }
                >
                  3
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID="commentButton"
              >
                <Icon
                  name="comment-outline"
                  size={46.690000000000005}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#505050",
                      "marginRight": "10%",
                    }
                  }
                />
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Object {
                      "color": "#505050",
                      "fontSize": 32.12572956370018,
                      "fontWeight": "bold",
                      "marginLeft": "5%",
                      "textAlign": "center",
                    }
                  }
                >
                  4
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff59b",
        "borderColor": "#104e01",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 106.72,
        "justifyContent": "center",
        "left": "80%",
        "opacity": 1,
        "position": "absolute",
        "top": "86%",
        "width": 106.72,
        "zIndex": 1,
      }
    }
    testID="filterButton"
  >
    <Icon
      name="filter"
      size={46.690000000000005}
      style={
        Object {
          "color": "#104e01",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
